i6_models/parts/conformer/convolution.py

```diff
@@ -0,0 +1,59 @@
+from torch import nn
+
+
+class ConformerConvolutionV1(nn.Module):
+    """
+    Conformer convolution module.
+    see also: https://github.com/espnet/espnet/blob/713e784c0815ebba2053131307db5f00af5159ea/espnet/nets/pytorch_backend/conformer/convolution.py#L13
+    """
+
+    def __init__(self, channels, kernel_size, dropout=0.1):
+        """
+        :param int channels: number of channels for conv layers
+        :param int kernel_size: kernel size of conv layers
+        """
+        super().__init__()
+
+        # kernel size has to be odd to get same input length when using odd strides. here we use stride 1.
```

Comment:
Why is that? Even with even kernel sizes, you can get the same input length.

Reply:
When using even kernel sizes, i think this would work when using padding `kernel_size / 2` but again you have issues with odd kernel sizes. I am not sure i tried to test this and i had issues with that.

Reply:
espnet also has this check: https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/conformer/convolution.py#L26

Reply:
With `padding = "same"`, you would add `(kernel_size - 1) // 2` padding to the left, and `ceildiv(kernel_size - 1, 2)` padding to the right.

Reply:
With even kernel sizes you can not do symmetric padding, this is why there is the `padding: "same"` option that handles this automatically. But I would also be fine with disallowing even Kernel sizes, as this is usually something not so nice to do when you want to keep the sequence length.

Reply:
Using `padding="same"` is just the same for odd kernel sizes as what we have now, and additionally allows even kernel sizes, even if they are maybe not so nice. I don't really understand why we artificially want to have such restriction. Esp if you want to do some hyper param search over kernel sizes.